packages/gestalt/src/HelpButton.js

```diff
@@ -0,0 +1,200 @@
+  const [isActive, setActive] = useState(false);
+  const popoverId = useId();
+  const {
+    accessibilityTooltipMessage,
+    accessibilityLinkLabel: accessibilityDefaultLinkLabel,
+    accessibilityIcon,
+  } = useDefaultLabelContext('HelpButton');
+
+  useEffect(() => {
+    if (isOpen) {
+      ref?.current?.focus();
+    }
+  }, [isOpen]);
+
+  const toggleView = () => {
+    setIsOpen((currVal) => !currVal);
+    setActive((currVal) => !currVal);
+  };
+
+  const onHandleTap = (props) => {
+    toggleView();
+    if (onClick) {
+      onClick(props);
+    }
+  };
+
+  const bgIconColor = isHovered || isActive || isFocused ? 'default' : 'subtle';
+
+  const tooltipZIndex = zIndex ?? new FixedZIndex(1);
+
+  const zIndexLayer = new CompositeZIndex([new FixedZIndex(tooltipZIndex.index())]);
+
+  const textElement = typeof text === 'string' ? <Text align="center">{text}</Text> : text;
+
+  return (
+    <Flex alignItems="center" justifyContent="center" flex="none">
+      <Tooltip
+        text={accessibilityTooltipMessage}
+        idealDirection={idealDirection}
+        zIndex={tooltipZIndex}
+      >
+        <TapArea
+          accessibilityExpanded={isOpen}
+          accessibilityControls={popoverId}
+          accessibilityLabel={accessibilityTooltipMessage}
+          fullWidth={false}
+          onTap={onHandleTap}
+          ref={ref}
+          role="button"
+          rounding="circle"
+          onBlur={() => {
+            setFocused(false);
+            setActive(false);
+          }}
+          onFocus={() => setFocused(true)}
+          onMouseDown={() => setActive(true)}
+          onMouseEnter={() => setHovered(true)}
+          onMouseLeave={() => setHovered(false)}
+          onMouseUp={() => setActive(false)}
+          tabIndex={0}
+        >
+          <Box
+            width={16}
+            height={16}
+            rounding="circle"
+            display="flex"
+            alignItems="center"
+            justifyContent="center"
+            dangerouslySetInlineStyle={{
+              __style: {
+                backgroundColor: `var(--color-text-${bgIconColor})`,
+              },
+            }}
+          >
+            <Icon
+              accessibilityLabel={accessibilityIcon}
+              color="inverse"
+              icon="question-mark"
+              size={8}
+            />
+          </Box>
+        </TapArea>
+      </Tooltip>
+      {isOpen && (
+        <Box zIndex={zIndexLayer}>
+          <Popover
+            id={popoverId}
+            accessibilityLabel={accessibilityPopoverLabel}
+            anchor={ref.current}
+            onDismiss={() => {
+              toggleView();
+              setActive(false);
+            }}
+            idealDirection={idealDirection ?? 'down'}
+            positionRelativeToAnchor
+          >
+            <Box
+              padding={4}
+              rounding={4}
+              minWidth={230}
+              onBlur={() => {
```

Comment:
this onBlur and onFocus are confusing. this Box is not an interactive element. have you tried console.lohgging within this events?

Reply:
Yep, the `tabIndex={0}` set this box container as a focusable element. To focus on the first element inner the dialog without a "interactive element" (DismissButton, for example) I set this prop on Box around the text and this element starts focusing.